Source/SwiftLintFramework/Rules/SwitchCaseAlignmentRule.swift

```diff
@@ -99,18 +45,124 @@ public struct SwitchCaseAlignmentRule: ASTRule, ConfigurationProviderRule {
 
         let caseLocations = caseStatements.compactMap { caseDict -> Location? in
             guard let byteOffset = caseDict.offset,
-                let (line, char) = contents.lineAndCharacter(forByteOffset: byteOffset) else {
-                    return nil
+                  let (line, char) = contents.lineAndCharacter(forByteOffset: byteOffset) else {
+                return nil
             }
+
             return Location(file: file.path, line: line, character: char)
         }
 
+        guard let firstCase = caseLocations.first,
+              let firstCaseCharacter = firstCase.character else {
+            return []
+        }
+
+        // If indent_cases is on, the first case should be indented from its containing switch.
+        if configuration.indentedCases, firstCaseCharacter <= switchCharacter {
+            return caseLocations.map(locationToViolation)
+        }
+
+        let indentation = configuration.indentedCases ? firstCaseCharacter - switchCharacter : 0
+
         return caseLocations
-            .filter { $0.character != switchCharacter }
-            .map {
-                StyleViolation(ruleDescription: type(of: self).description,
-                               severity: configuration.severity,
-                               location: $0)
-            }
+            .filter { $0.character != switchCharacter + indentation }
+            .map(locationToViolation)
+    }
+
+    private func locationToViolation(_ location: Location) -> StyleViolation {
+        return StyleViolation(ruleDescription: type(of: self).description,
+                              severity: configuration.severityConfiguration.severity,
+                              location: location)
+    }
+}
+
+public extension SwitchCaseAlignmentRule {
```

Comment:
this probably should be `internal`

Reply:
@marcelofabri I actually think this has to remain public or the Test target won't be able to access the examples.

I'm defining the ACL on the extension instead of the internal struct itself.

Reply:
You can use `@testable import SwiftLintFramework` on the test class

Reply:
Oh duh! Done.